modules/onboarding/src/main/kotlin/edu/stanford/spezi/module/onboarding/spezi/flow/OnboardingNavigationPath.kt

```diff
@@ -0,0 +1,96 @@
+package edu.stanford.spezi.module.onboarding.spezi.flow
+
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.CompositionLocalProvider
+import androidx.navigation.NavController
+import androidx.navigation.NavGraph
+import androidx.navigation.NavType
+import androidx.navigation.compose.composable
+import androidx.navigation.createGraph
+import androidx.navigation.navArgument
+import androidx.navigation.navOptions
+import edu.stanford.spezi.core.utils.UUID
+
+class OnboardingNavigationPath internal constructor(
+    internal val navController: NavController,
+    internal var startDestination: String,
+    internal var steps: List<OnboardingNavigationStep>,
+) {
+    private val customSteps = mutableListOf<OnboardingNavigationStep>()
+
+    private val currentOnboardingStepId: String? get() {
+        val currentQueue = navController.backQueue
+        val lastStep = currentQueue.lastOrNull { queueItem ->
+            !(queueItem.destination.route ?: "").startsWith(customRoute(""))
+        }
+
+        return lastStep?.destination?.route ?: steps.firstOrNull()?.id
+    }
+
+    fun nextStep() {
+        val currentOnboardingStepId = currentOnboardingStepId
+        val currentStepIndex = steps.indexOfFirst { it.id == currentOnboardingStepId }
+        if (currentStepIndex < 0 || currentStepIndex + 1 >= steps.size) {
+            return
+        }
+
+        navController.navigate(
+            route = steps[currentStepIndex + 1].id,
+            navOptions = navOptions {}, // TODO: Anything important/relevant needed here?
+            navigatorExtras = null, // TODO: Anything important/relevant needed here?
+        )
+    }
+
+    fun append(id: String) {
+        val step = steps.firstOrNull { it.id == id } ?: error("")
+
+        navController.navigate(
```

Comment:
why do we attempt to navigate when appending? This won't work if the route hasn't been registered to the graph first, same for the append method below